Class#singleton_class needs a feature to create a singleton class when it does not have one

I found that current `Class#singleton_class` causes panic when applying to objects other than classes:

```ruby
Goby 0.1.7 😍 😺 🤤
» "a".singleton_class
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x43d1b35]

goroutine 1 [running]:
github.com/goby-lang/goby/vm.(*RClass).toString(0x0, 0xc4201c2a80, 0xffffffffffffffff)
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/vm/class.go:1205 +0x5
github.com/goby-lang/goby/vm.(*VM).GetREPLResult(0xc4200a6820, 0xc42000ea88, 0x1)
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/vm/repl.go:73 +0x4e
github.com/goby-lang/goby/igb.StartIgb(0x4577b0d, 0x5)
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/igb/repl.go:233 +0xd34
main.main()
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/goby.go:29 +0x954
```

According to http://devdocs.io/ruby~2.5/object#method-i-singleton_class:

> Returns the singleton class of obj. This method creates a new singleton class if obj does not have one.

The feature to create a new singleton class is missing in current Goby.

--- a/vm/class.go
+++ b/vm/class.go
@@ -1024,6 +1024,13 @@ func builtinClassCommonInstanceMethods() []*BuiltinMethodObject {
 			Name: "singleton_class",
 			Fn: func(receiver Object, sourceLine int) builtinMethodBody {
 				return func(t *thread, args []Object, blockFrame *normalCallFrame) Object {
+					r := receiver
+					if r.SingletonClass() == nil {
+						id := t.vm.initIntegerObject(r.id())
+						singletonClass := t.vm.createRClass(fmt.Sprintf("#<Class:#<%s:%s>>", r.Class().Name, id.toString()))
+						singletonClass.isSingleton = true
+						return singletonClass
+					}
 					return receiver.SingletonClass()
 				}
 			},

--- a/vm/class_test.go
+++ b/vm/class_test.go
@@ -1100,6 +1100,12 @@ func TestClassSingletonClassMethod(t *testing.T) {
 		class Foo < Bar; end
 		Foo.singleton_class.superclass.name
 		`, "#<Class:Bar>"},
+		// Check if this works on non-class objects
+		{`'a'.singleton_class.to_s.slice(1..16).to_s`, "<Class:#<String:"},
+		{`1.singleton_class.to_s.slice(1..17).to_s`, "<Class:#<Integer:"},
+		{`nil.singleton_class.to_s.slice(1..14).to_s`, "<Class:#<Null:"},
+		{`[1,2].singleton_class.to_s.slice(1..15).to_s`, "<Class:#<Array:"},
+		{`{key: "value"}.singleton_class.to_s.slice(1..14).to_s`, "<Class:#<Hash:"},
 	}
 
 	for i, tt := range tests {

